packages/compiler/src/server/lib-provider.ts

```diff
@@ -1,54 +1,57 @@
 import { joinPaths } from "../core/path-utils.js";
 import { NpmPackage, NpmPackageProvider } from "./npm-package-provider.js";
 
-export class EmitterProvider {
-  private isEmitterPackageCache = new Map<string, boolean>();
-  constructor(private npmPackageProvider: NpmPackageProvider) {}
+export class LibraryProvider {
+  private isLibPackageCache = new Map<string, boolean>();
+  constructor(
+    private npmPackageProvider: NpmPackageProvider,
+    private filter: (obj: Record<string, any>) => boolean,
+  ) {}
 
   /**
    *
-   * @param startFolder folder starts to search for package.json with emitters defined as dependencies
+   * @param startFolder folder starts to search for package.json with emitters/linters defined as dependencies
```

Comment:
the comment should be general too. There shouldn't be any emitter or linter concept in this class any more.

Reply:
same comment for below